ghost/admin/app/components/gh-context-menu.js

```diff
@@ -2,16 +2,68 @@ import Component from '@glimmer/component';
 import SelectionList from '../utils/selection-list';
 import {action} from '@ember/object';
 import {inject as service} from '@ember/service';
+import {task} from 'ember-concurrency';
 import {tracked} from '@glimmer/tracking';
 
+const states = [
+    'default', // default state
+    'open', // menu open
+    'modal', // modal open
+    'loading' // performing an action
+];
+
 export default class GhContextMenu extends Component {
     @service dropdown;
+    @service modals;
 
     @tracked isOpen = false;
     @tracked left = 0;
     @tracked top = 0;
     @tracked selectionList = new SelectionList();
 
+    state = states[0];
+
+    #originalConfirm = null;
+    #modal = null;
+
+    setState(state) {
+        switch (state) {
+        case this.state:
+            return;
+        case 'default':
+            this.isOpen = false;
+            this.selectionList.unfreeze();
+            this.#closeModal();
+            this.state = state;
+            return;
+        case 'open':
+            if (this.state !== 'default') {
+                return;
+            }
+            this.isOpen = true;
+            this.selectionList.freeze();
+            this.#closeModal();
+            this.state = state;
+            return;
+        case 'modal':
+            if (this.state !== 'open') {
+                return;
+            }
+            this.isOpen = false;
+            this.selectionList.freeze();
+            this.state = state;
+            return;
+        case 'loading':
+            if (this.state !== 'open' && this.state !== 'modal') {
+                return;
+            }
+            this.isOpen = false;
+            this.selectionList.freeze();
+            this.state = state;
+            return;
```

Comment:
I'm wondering if we should already close here. Because there is no indication that the action is still loading at the moment, unless we add that somewhere. But probably easier to add an indication in the context menu until the loading is done. Wdyt?